tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc

```diff
@@ -82,6 +88,32 @@ class TRTEngineOpTestBase : public OpsTestBase {
                                                          std::string(kOpName)));
     TF_ASSERT_OK(flib_def_->AddLibrary(graph->flib_def()));
 
+    string segment_string;
+    if (static_engine) {
+      convert::ConversionParams params;
+      convert::EngineInfo info;
+      info.segment_graph_def.CopyFrom(graph_def);
+      info.precision_mode = TrtPrecisionMode::FP32;
+      info.max_workspace_size_bytes = 1 << 20;
+      info.engine_name = "TRTEngineOP_0_0";
+      params.use_implicit_batch = use_implicit_batch;
+      params.trt_logger_name = "DefaultLogger";
+
+      TrtShapeOptimizationProfile profile;
+      TensorShape my_shape;
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{4, 2}, &my_shape));
+      profile.AddShape({my_shape, {}});
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{1, 2}, &my_shape));
```

Comment:
Currently, we only have tests with input shape {1,2}. For this reason, shall we remove {4,2}?
Also, shall we also add a test that has an input shape rather then these shapes used to prebuild the engine?

Reply:
We should not remove the unused profile. It is there to test profile selection: we have {4,2} as profile zero, to run the test TRTEngineOp has to select profile 1. I have added a comment explaining that.

Good point about testing with an input for which we do not have an engine. I assume that native segment fall-back is the expected behavior. I have corrected GetEngine and added a test.